vnext/template/cpp-app/src/App.cpp

```diff
@@ -36,6 +36,10 @@ App::App() noexcept
 #else
     InstanceSettings().UseDeveloperSupport(false);
 #endif
+{{#useHermes}}
```

Comment:
It would be better if the project file defines macros depending on the BuildFlags.props and then we sue these macros here.
This way developers can change the BuildFlags.props to make the switch.
Otherwise, it makes it really confusing: the BuildFlags.props file is there, but changing it is not enough. One needs to know how to modify the code in some "secret" places.

Reply:
The same technique can be used for C# projects.

---
In reply to: [513128016](https://github.com/microsoft/react-native-windows/pull/6350#discussion_r513128016) [](ancestors = 513128016)

Reply:
I can see how this could be confusing. Do you think this would be clearer if we renamed UseHermes/UseV8 to something like AllowHermes/AllowV8?

I think allowing the secanrio of building a version of MSRN supporting either engine is useful, so I'm not sure we should couple allowing Hermes + choosing Hermes.

Annoyingly, we do have a third step beyond the build flag + instance settings of adding the NuGet package, which is currently too large to download by default. Will be nice to have a thinner package so we can do that. I'm hoping the "--useHermes" project generation flag can make help simplify things a bit, but that only really helps new projects. I plan to update docs with new steps to add to an existing project though.

Reply:
There's actually a separate change where we want to do Hermes specific things in the build (generate bytecode instead of JS) when using Hermes. Will take a look at this again, since we will want a definitive source of choice there anyway.

Reply:
Changed to now set the default JSI engine override based on preprocessor definition. Can still be overridden though.